rules/integrations/azure/impact_azure_compute_vm_snapshot_deletion.toml

```diff
@@ -0,0 +1,132 @@
+- DevOps automation tools might delete temporary snapshots created during deployment or testing processes. Identify service principals used by CI/CD pipelines and consider time-based exceptions during deployment windows.
+- Storage optimization initiatives may involve deleting old or redundant snapshots to reduce costs. Coordinate with infrastructure teams to understand planned optimization activities and create exceptions during documented maintenance windows.
+- Disaster recovery testing may involve creating and deleting test snapshots. Work with business continuity teams to identify these patterns and create exceptions during scheduled DR testing periods.
+
+### Response and remediation
+
+- Immediately investigate whether the deletion was authorized by verifying with the account owner, backup administrators, or relevant stakeholders.
+- If the deletion was unauthorized, disable the compromised user account or service principal immediately to prevent further damage.
+- Check if the snapshot can be recovered through Azure backup services or soft-delete capabilities if enabled.
+- Create new snapshots of critical disks immediately if the deleted snapshot was part of your backup strategy.
+- Review and audit all Azure RBAC permissions to identify how the attacker gained snapshot deletion capabilities.
+- Conduct a full security assessment to identify the initial access vector and any other compromised accounts or resources.
+- Implement Azure Resource Locks on critical snapshots to prevent accidental or malicious deletion.
+- Configure Azure Policy to restrict snapshot deletion permissions to only authorized backup administrators.
+- Enable Azure Activity Log alerts to notify security teams immediately when snapshots are deleted.
+- Review backup and disaster recovery procedures to ensure redundant backup mechanisms exist beyond Azure snapshots.
+- Document the incident and update security policies and procedures to prevent similar incidents in the future.
+"""
+references = [
+    "https://www.microsoft.com/en-us/security/blog/2025/08/27/storm-0501s-evolving-techniques-lead-to-cloud-based-ransomware/",
+]
+risk_score = 21
+rule_id = "c3d4e5f6-7a8b-9c0d-1e2f-3a4b5c6d7e8f"
+severity = "low"
+tags = [
+    "Domain: Cloud",
+    "Domain: Storage",
+    "Data Source: Azure",
+    "Data Source: Azure Activity Logs",
+    "Use Case: Threat Detection",
+    "Tactic: Impact",
+    "Resources: Investigation Guide",
+]
+timestamp_override = "event.ingested"
+type = "new_terms"
+
+query = '''
+event.dataset: azure.activitylogs and
+    azure.activitylogs.operation_name: "MICROSOFT.COMPUTE/SNAPSHOTS/DELETE" and
+    azure.activitylogs.properties.status_code: "Accepted" and
+    azure.activitylogs.identity.claims_initiated_by_user.name: *
```

Comment:
Is there ever an instance where this field isn't populated? 
Also, is `azure.activitylogs.result_type: Accepted` equivalent to `event.outcome: success`? If so should you add this to the query to exclude failed attemps?